src/main/scala/com/gu/kindlegen/Lambda.scala

```
@@ -1,43 +1,63 @@
 package com.gu.kindlegen
 
-import com.amazonaws.services.lambda.runtime.Context
-/**
- * This is compatible with aws' lambda JSON to POJO conversion.
- * You can test your lambda by sending it the following payload:
- * {"name": "Bob"}
- */
-class LambdaInput() {
-  var name: String = _
-  def getName(): String = name
-  def setName(theName: String): Unit = name = theName
-}
+import java.nio.file.Files
+import java.time.{Instant, LocalDate}
+import java.time.ZoneOffset.UTC
 
-case class Env(app: String, stack: String, stage: String) {
-  override def toString: String = s"App: $app, Stack: $stack, Stage: $stage\n"
-}
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
 
-object Env {
-  def apply(): Env = Env(
-    Option(System.getenv("App")).getOrElse("DEV"),
-    Option(System.getenv("Stack")).getOrElse("DEV"),
-    Option(System.getenv("Stage")).getOrElse("DEV")
-  )
-}
+import awscala._
+import awscala.s3._
+import com.amazonaws.services.lambda.runtime.Context
 
 object Lambda {
+  // env.AWS_REGION -> eu-west-1
+  // env.Stage -> CODE
 
   /*
    * This is your lambda entry point
    */
-  def handler(lambdaInput: LambdaInput, context: Context): Unit = {
+  def handler(parameters: java.util.Map[String, Any], context: Context): Unit = {
     val logger = context.getLogger
-    val env = Env()
-    logger.log(s"Starting $env")
-    logger.log(process(lambdaInput.name, env))
+
+    val params = parameters.asScala
+    val date =
+      params.get("date").map(_.toString).map(LocalDate.parse)
+        .orElse(params.get("time").map(_.toString).map(Instant.parse(_).atZone(UTC).toLocalDate))  // scheduled event
+        .getOrElse(LocalDate.now)
+
+    val stage = sys.env("Stage")  // CODE or PROD
```

If we do not have particular config we need that we expect to receive from environment variables then we should fail loudly. E.g

```
Option(sys.env("AWS_REGION")) getOrElse sys.error("")
```

---

Will do.

---

I've removed the quoted line and made sure we have appropriate defaults for other properties.